Userland/Applications/Browser/Tab.cpp

```diff
@@ -397,6 +414,38 @@ Tab::Tab(Type type)
     js_console_action->set_status_tip("Open JavaScript console for this page");
     inspect_menu.add_action(js_console_action);
 
+    auto& settings_menu = m_menubar->add_menu("&Settings");
+
+    m_search_engine_actions.set_exclusive(true);
+    auto& search_engine_menu = settings_menu.add_submenu("&Search Engine");
+
+    auto add_search_engine = [&](auto& name, auto& url_format) {
+        auto action = GUI::Action::create_checkable(
+            name, [&](auto&) {
+                dbgln("Setting search engine to {}", url_format);
+                s_search_engine_format = url_format;
+            },
+            this);
+        search_engine_menu.add_action(action);
+        m_search_engine_actions.add_action(action);
+    };
+
+    auto disable_search_engine_action = GUI::Action::create_checkable(
+        "Disable", [this](auto&) {
+            dbgln("Disabling search engine");
+            s_search_engine_format = "{}";
+        },
+        this);
+    search_engine_menu.add_action(disable_search_engine_action);
+    m_search_engine_actions.add_action(disable_search_engine_action);
+    disable_search_engine_action->set_checked(true);
+
+    // FIXME: Support adding custom search engines
+    add_search_engine("Bing", "https://www.bing.com/search?q={}");
+    add_search_engine("DuckDuckGo", "https://duckduckgo.com/?q={}");
+    add_search_engine("Google", "https://google.com/search?q={}");
+    add_search_engine("Wikipedia", "https://en.wikipedia.org/w/index.php?search={}");
```

Comment:
We cant actually load wikipedia at the moment (As we dont support any modern TLS1.2/1.3 cipher suites), so having it as an option right now is not very helpful